packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -48,64 +51,108 @@ contract MixinExchangeCore is
     // Orders with a salt less than their maker's epoch are considered cancelled
     mapping (address => uint256) public makerEpoch;
 
-    /// @dev Cancels all orders reated by sender with a salt less than or equal to the specified salt value.
-    /// @param salt Orders created with a salt less or equal to this value will be cancelled.
-    function cancelOrdersUpTo(uint256 salt)
-        external
-    {
-        uint256 newMakerEpoch = salt + 1;  // makerEpoch is initialized to 0, so to cancelUpTo we need salt + 1
-        require(
-            newMakerEpoch > makerEpoch[msg.sender],  // epoch must be monotonically increasing
-            INVALID_NEW_MAKER_EPOCH
-        );
-        makerEpoch[msg.sender] = newMakerEpoch;
-        emit CancelUpTo(msg.sender, newMakerEpoch);
-    }
+    ////// Core exchange functions //////
 
-    /// @dev Fills the input order.
-    /// @param order Order struct containing order specifications.
-    /// @param takerAssetFillAmount Desired amount of takerAsset to sell.
-    /// @param signature Proof that order has been created by maker.
-    /// @return Amounts filled and fees paid by maker and taker.
-    function fillOrder(
-        Order memory order,
-        uint256 takerAssetFillAmount,
-        bytes memory signature)
+    /// @dev Gets information about an order.
+    /// @param order Order to gather information on.
+    /// @return status Status of order. Statuses are defined in the LibStatus.Status struct.
+    /// @return orderHash Keccak-256 EIP712 hash of the order.
+    /// @return takerAssetFilledAmount Amount of order that has been filled.
+    function getOrderInfo(Order memory order)
```

Comment:
Knowing the amount remaining to be filled is needed in the forwarding contract. Is there an alternative? In the mean time I have been keeping getUnavailable around in my PR.

Reply:
@dekz  - I chatted offline with @abandeali1 about this & we decided to keep it public. Hoping it can be useful in the forwarding contract, for getting amount to fill & order state.